Mathlib/RingTheory/PowerSeries/WellKnown.lean

```diff
@@ -64,6 +65,87 @@ theorem map_invUnitsSub (f : R →+* S) (u : Rˣ) :
 
 #align power_series.map_inv_units_sub PowerSeries.map_invUnitsSub
 
+section invOneSubPow
+
+variable {S : Type _} [CommRing S] (d : ℕ)
+
+theorem mk_one_pow_eq_mk_choose_add :
+    (@mk S 1) ^ (d + 1) = @mk S fun n => Nat.choose (d + n) d := by
+  induction d with
+  | zero => simp only [Nat.zero_eq, zero_add, pow_one, Nat.choose_zero_right, Nat.cast_one]; rfl
+  | succ d hd =>
+      rw [pow_add, pow_one, mul_comm, show Nat.succ d = d + 1 by rfl, ext_iff]; exact fun n => by
+        rw [hd, coeff_mul]; simp only [coeff_mk, one_mul]; rw [Nat.succ_add, Nat.choose_succ_succ,
+        ← Finset.sum_antidiagonal_choose_add]; simp only [Pi.one_apply, one_mul, Nat.cast_sum]
```

Comment:
This is long proof disguised as 3 lines using `;`. Here's a proof of about the same length which is easier to understand.
```
  | succ d hd =>
      ext n
      -- step 1: get rid of succ, use induction hypothesis, expand out
      rw [Nat.succ_eq_add_one, pow_add, hd, pow_one, mul_comm, coeff_mul]
      -- Now tidy up the mess `coeff_mul` made
      simp only [coeff_mk, Pi.one_apply, one_mul]
      -- Now cast down from `S` to `ℕ`.
      norm_cast
      congr
      -- now the key lemma rewrites the sensible way
      rw [Finset.sum_antidiagonal_choose_add]
      -- and now it's easy from a standard fact about binomial coefficients
      rw [← Nat.choose_succ_succ]
      congr
      -- ... (and commutativity and associativity of addition)
      exact add_right_comm d n 1
```